### PR TITLE
planning: (1) update the functio IsOnLane() and IsOnRoad() in Referen…

### PR DESCRIPTION
**Pull request by jmtao**
…ceLine; (2) fix the check on sideways/medians for crosswalk (#1590)

### CODE REVIEW

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 1]
```diff
@@ -263,10 +277,10 @@ void Crosswalk::MakeDecisions(Frame* const frame,
 
       if (stop) {
         pedestrians.push_back(obstacle_id);
-        ADEBUG << "wait for: obstacle_id[" << obstacle_id << "] type["
+        AERROR << "wait for: obstacle_id[" << obstacle_id << "] type["
```
Comment (startcode) [inline modules/planning/tasks/traffic_decider/crosswalk.cc]: Seems this one should be ADEBUG

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 2]
```diff
@@ -229,14 +242,15 @@ void Crosswalk::MakeDecisions(Frame* const frame,
         if (is_path_cross) {
           stop = true;
         }
-        ADEBUG << "need_stop(between 11 & l2): obstacle_id[" << obstacle_id
+        AERROR << "need_stop(between 11 & l2): obstacle_id[" << obstacle_id
```
Comment (startcode) [inline modules/planning/tasks/traffic_decider/crosswalk.cc]: ADEBUG?

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 3]
```diff
@@ -156,69 +156,82 @@ void Crosswalk::MakeDecisions(Frame* const frame,
...
+                Vec2d(reference_line_info->AdcPlanningPoint().path_point().x(),
+                      reference_line_info->AdcPlanningPoint().path_point().y());
+            const auto ovstacle_position =
+                Vec2d(perception_obstacle.position().x(),
+                      perception_obstacle.position().y());
+            auto obs_to_adc = adc_path_point - ovstacle_position;
+            const double kEpsilon = 1e-6;
+            if (obstacle_v.InnerProd(obs_to_adc) > kEpsilon) {
+              stop = true;
+              AERROR << "need_stop(<=11): obstacle_id[" << obstacle_id
```
Comment (startcode) [inline modules/planning/tasks/traffic_decider/crosswalk.cc]: ADEBUG?

File: modules/planning/tasks/traffic_decider/crosswalk.cc [hunk 4]
```diff
@@ -156,69 +156,82 @@ void Crosswalk::MakeDecisions(Frame* const frame,
...
+                << "] adc_end_edge_s[ " << adc_end_edge_s
+                << "] crosswalk_id[" << crosswalk_id << "] ON_ROAD";
+          }
+        } else {
+          // (3) when l_distance <= strick_l_distance
+          //     + NOT on_road(i.e. on crosswalk/median etc)
+          //     STOP if paths cross
+          if (is_path_cross) {
+            stop = true;
+            AERROR << "need_stop(<=11): obstacle_id["
```
Comment (startcode) [inline modules/planning/tasks/traffic_decider/crosswalk.cc]: ADEBUG?

File: modules/planning/tasks/traffic_decider/crosswalk.cc (hunk 1 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: fixed

File: modules/planning/tasks/traffic_decider/crosswalk.cc (hunk 2 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: fixed

File: modules/planning/tasks/traffic_decider/crosswalk.cc (hunk 3 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: fixed

File: modules/planning/tasks/traffic_decider/crosswalk.cc (hunk 4 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/crosswalk.cc]: fixed

**Pull request merged**